### PR TITLE
CEL Interperter for Conditions

### PR DESCRIPTION
**Pull request by chirino**
Continues on #254 in a way that is backward compatible.  To use cel based conditions, you must prefix your condition with `cel:`

### CODE REVIEW

File: limitador/src/limit.rs [hunk 1]
```diff
@@ -991,11 +1055,86 @@ mod tests {
...
+            assert!(limit.applies(&values));
+        }
+
+        #[test]
+        fn vars_var() {
+            let values = HashMap::from([("vars".to_string(), "hello".to_string())]);
+
+            // we can't access simple variables that conflict with built-ins (the vars map)
+            let limit = limit_with_condition(vec![r#"cel:   vars == "hello"     "#]);
+            assert!(!limit.applies(&values));
```
Comment (alexsnaps) [inline limitador/src/limit.rs:1133]: Real minor neat, but the `!` I find so easy to miss (tho you did put a clear comment there), _and_ clippy will complain about `assert_eq!(cond, false)`, what I recently started doing for these `assert!(!false)`, is that I use `std::ops::Not` and try to express these as `assert!(limit.applies(&values).not())` instead. Now don't feel you need to use the same pattern, just sharing…

Comment (chirino) [reply limitador/src/limit.rs:1133]: I miss it all the time.. kinda wish there was an `assert_not!`

File: limitador/src/limit.rs [hunk 2]
```diff
@@ -886,6 +886,14 @@ mod conditions {
 mod tests {
     use super::*;
 
+    macro_rules! assert_false {
+        ($cond:expr $(,)?) => {
+            paste::item! {
+               assert!(!$cond)
```
[the remark this answers is not in the document]
Comment (chirino) [reply limitador/src/limit.rs]: sounds good to me.  updated.

File: limitador/Cargo.toml [hunk 3]
```diff
@@ -33,6 +34,8 @@ async-trait = "0.1"
 cfg-if = "1"
 tracing = "0.1.40"
 metrics = "0.22.3"
+cel-parser = "0.6.0"
+cel-interpreter = "0.5.0"
```
[the remark this answers is not in the document]
Comment (chirino) [reply limitador/Cargo.toml]: updated.

**Comment by alexsnaps** [comment]
Closing this in favor of #396 which drops support for all the legacy syntax altogether

**Pull request closed without merging**